databroker/core.py

```diff
@@ -1443,6 +1460,50 @@ def parse_handler_registry(handler_registry):
     return result
 
 
+def load_transforms(transforms):
+    """
+    Parse mapping of spec name to 'import path' into mapping to class itself.
+
+    Parameters
+    ----------
+    transforms : dict or None
+        Values may be string 'import paths' to classes or actual classes.
+
+    Examples
+    --------
+    Pass in name; get back actual class.
+
+    >>> load_transforms({'descriptor': 'package.module.ClassName'})
+    {'descriptor': <package.module.ClassName>}
+
+    """
+    transformable = {'start', 'stop', 'resource', 'descriptor'}
+
+    if transforms is None:
+        result = {key: lambda doc: doc for key in transformable}
+        return result
+    elif isinstance(transforms, dict):
```

Comment:
I hesitate to even mention this and wouldn't block merging on it, but in the spirit of embracing duck typing, it would be fine to just assume that, if `transforms` is not None, it is a dict-like and proceed. IMO the cost of restricting users to strict `dict` subclasses outweighs the benefit of trying to give a more helpful error message in the edge case where the user accidentally gives us some object here that we cannot use.

Reply:
What type of objects that are not dict or None would also be acceptable?

Reply:
User-defined dict-like objects would be OK, such as might arise from custom config-parsing code. Or these...of which some are `dict` subclasses and some are not.

```
In [11]: import collections                                                                                                                                                                   

In [12]: isinstance(collections.OrderedDict(), dict)                                                                                                                                          
Out[12]: True

In [13]: isinstance(collections.ChainMap(), dict)                                                                                                                                             
Out[13]: False

In [14]: isinstance(collections.defaultdict(lambda: (lambda: x)), dict)                                                                                                                       
Out[14]: True

In [19]: isinstance(collections.UserDict(), dict)                                                                                                                                             
Out[19]: False
```

Rather than simply giving up on validation, as I originally suggested here, it may be better to validate against an abstract base class.

```py
In [15]: isinstance(collections.OrderedDict(), collections.abc.Mapping)                                                                                                                       
Out[15]: True

In [16]: isinstance(collections.ChainMap(), collections.abc.Mapping)                                                                                                                          
Out[16]: True

In [17]: isinstance(collections.defaultdict(lambda: (lambda: x)), collections.abc.Mapping)                                                                                                    
Out[17]: True

In [18]: isinstance(collections.UserDict(), collections.abc.Mapping)                                                                                                                          
Out[18]: True
```

Note that none of these types _actually_ subclass the type `collections.abc.Mapping`, but `collections.abc.Mapping` uses [`__subclasshook__`](https://docs.python.org/3/library/abc.html#abc.ABCMeta.__subclasshook__) to check if an object has the expected interface (`__getitem__`, `__iter__`, etc.) and then tells `isinstance` what to return.